packages/react-dismissible/src/Dismissible.tsx

```diff
@@ -47,64 +48,63 @@ export interface Props {
   document?: Document
 }
 
-export class Dismissible extends React.Component<Props, {}> {
-  area: any
-  get documents() {
-    const targets = [document]
-
-    if (this.props.document) {
-      targets.push(this.props.document)
-    }
-    return targets
-  }
-  componentDidMount() {
-    if (this.props.click) {
-      this.documents.forEach(document =>
-        document.body.addEventListener('click', this.handleDocumentClick)
-      )
-    }
-
-    if (this.props.escape) {
-      this.documents.forEach(document =>
-        document.addEventListener('keydown', this.handleEscape)
-      )
+export const Dismissible: React.FC<Props> = ({
+  onDismiss,
+  escape = false,
+  click = false,
+  disabled = false,
+  document: customDocument,
+  ...props
+}) => {
+  const area: any = useRef()
+  const documents: any[] = customDocument
+    ? [document, customDocument]
+    : [document]
+
+  useEffect(() => {
+    const stopAndPrevent = (event: MouseEvent) => {
+      event.stopPropagation()
+      event.stopImmediatePropagation()
+      event.preventDefault()
     }
-  }
 
-  componentWillUnmount() {
-    this.documents.forEach(document =>
-      document.body.removeEventListener('click', this.handleDocumentClick)
-    )
-    this.documents.forEach(document =>
-      document.removeEventListener('keydown', this.handleEscape)
-    )
-  }
+    const handleDocumentClick = (event: MouseEvent) => {
+      if (disabled) return
 
-  handleDocumentClick = (event: MouseEvent) => {
-    if (this.props.disabled) return
+      if (!area.current.contains(event.target)) {
+        stopAndPrevent(event)
+        onDismiss(event)
+      }
+    }
 
-    const area: any = this.area
+    const handleEscape = (event: KeyboardEvent) => {
+      if (disabled) return
 
-    if (!area.contains(event.target)) {
-      this.props.onDismiss(event)
-      event.stopPropagation()
-      event.stopImmediatePropagation()
-      event.preventDefault()
+      if (event.keyCode === 27) {
+        event.stopPropagation()
+        onDismiss(event)       
+      }
     }
-  }
 
-  handleEscape = (event: KeyboardEvent) => {
-    if (this.props.disabled) return
-
-    if (event.keyCode == 27) {
-      this.props.onDismiss(event)
-      event.stopPropagation()
+    if (click) {
+      documents.forEach(document =>
+        document.body.addEventListener('click', handleDocumentClick)
+      )
     }
-  }
 
-  render() {
-    const { onDismiss, click, escape, ...props } = this.props
+    if (escape) {
+      documents.forEach(document =>
+        document.addEventListener('keydown', handleEscape)
+      )
+    }
+    // Clean up event listeners on unmount
+    return () => {
+      documents.forEach(document => {
+        document.body.removeEventListener('click', handleDocumentClick)
+        document.removeEventListener('keydown', handleEscape)
+      })
+    }
+  }, [click, documents, escape, disabled, onDismiss])
```

Comment:
A new `documents` array is created on every render, which means the effect would rerun on every render. It needs to be the two values listed individually to prevent rerunning.
```suggestion
  }, [click, document, customDocument, escape, disabled, onDismiss])
```